packages/devtools_extensions/lib/src/template/_simulated_devtools_environment/_simulated_devtools_environment.dart

```diff
@@ -29,25 +31,40 @@ class SimulatedDevToolsWrapper extends StatefulWidget {
   const SimulatedDevToolsWrapper({
     super.key,
     required this.child,
-    required this.connected,
   });
 
   final Widget child;
 
-  final bool connected;
-
   @override
   State<SimulatedDevToolsWrapper> createState() =>
       _SimulatedDevToolsWrapperState();
 }
 
-class _SimulatedDevToolsWrapperState extends State<SimulatedDevToolsWrapper> {
+class _SimulatedDevToolsWrapperState extends State<SimulatedDevToolsWrapper>
+    with AutoDisposeMixin {
   late final _SimulatedDevToolsController simController;
 
+  late ConnectedState connectionState;
```

Comment:
It seems connectionState should be final. Or, we should initialize it in something like 'disWidgetChange'

Reply:
this is late because it is initialized in `initState` and it cannot be final because we update it in a listener.